devbox.go

```diff
@@ -214,7 +214,11 @@ func (d *Devbox) Shell() error {
 		if err != nil {
 			return err
 		}
-		opts = append(opts, nix.WithEnvVariables(env))
+		opts = append(
+			opts,
+			nix.WithEnvVariables(env),
+			nix.WithPath(filepath.Join(d.srcDir, ".devbox/conf/bin")),
```

Comment:
The name `WithPath` made me think this was setting the path to Nix. What do you think of having the caller update the `PATH` and then pass it in via `WithEnvVars`? That would also fix the inconsistency with how we currently bypass the "with-ed" env vars when reading the `PATH`.

Reply:
Actually, this wont work. We need to explicitly prepend the conf bin path in a few places so I think it has to be its own argument. If we just send an updated PATH then we'll end up prepending the original nix binaries (via the profile path). 

I do think we can clean stuff up by maybe just adding the config bin path by default (the way we do with profile bin dir). I think I'll do that instead.